src/terminator/mod.rs

```diff
@@ -313,6 +313,10 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
                 )?;
 
                 let arg_locals = self.frame().mir.args_iter();
+                // FIXME: impl ExactSizeIterator and use args_locals.len()
+                // FIXME: last-use-in-cap-clause works by chance, insert some arguments and it will fail
+                // we currently write the first argument (unit) to the return field (unit)
```

Comment:
Are you sure we do this? I don't see anything writing to the return lvalue. `mir.args_iter()` doesn't include the return lvalue.